provision/kubernetes/job.go

```diff
@@ -234,44 +235,51 @@ func (p *kubernetesProvisioner) DestroyJob(ctx context.Context, job *jobTypes.Jo
 	return client.BatchV1().CronJobs(namespace).Delete(ctx, job.Name, metav1.DeleteOptions{})
 }
 
-func (p *kubernetesProvisioner) podsForJobs(ctx context.Context, client *ClusterClient, jobs []*jobTypes.Job) ([]apiv1.Pod, error) {
-	podList := []apiv1.Pod{}
-	for _, j := range jobs {
-		labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{tsuruLabelJobName: j.Name}}
-		listOptions := metav1.ListOptions{
-			LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
-		}
-		namespace := client.PoolNamespace(j.Pool)
-
-		pods, err := client.CoreV1().Pods(namespace).List(ctx, listOptions)
-		if err != nil {
-			return podList, err
-		}
-		podList = append(podList, pods.Items...)
+func (p *kubernetesProvisioner) getPodsForJob(ctx context.Context, client *ClusterClient, job *batchv1.Job) ([]apiv1.Pod, error) {
+	labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{"job-name": job.Name}}
+	listOptions := metav1.ListOptions{
+		LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
+	}
+	pods, err := client.CoreV1().Pods(job.Namespace).List(ctx, listOptions)
+	if err != nil {
+		return nil, err
 	}
-	return podList, nil
+	return pods.Items, nil
 }
 
-func (p *kubernetesProvisioner) podsToJobUnits(ctx context.Context, client *ClusterClient, pods []apiv1.Pod, job *jobTypes.Job) ([]provision.Unit, error) {
-	if len(pods) == 0 {
+func (p *kubernetesProvisioner) jobsToJobUnits(ctx context.Context, client *ClusterClient, k8sJobs []batchv1.Job, job *jobTypes.Job) ([]provision.Unit, error) {
+	if len(k8sJobs) == 0 {
 		return nil, nil
 	}
 	var units []provision.Unit
-	for _, pod := range pods {
+	for _, k8sJob := range k8sJobs {
 		var status provision.Status
-		if pod.Status.Phase == apiv1.PodRunning {
-			status, _ = extractStatusAndReasonFromContainerStatuses(pod.Status.ContainerStatuses)
-		} else {
-			status = stateMap[pod.Status.Phase]
+		var restarts *int32
+		pods, err := p.getPodsForJob(ctx, client, &k8sJob)
+		if err != nil {
+			return nil, err
+		}
+		for _, pod := range pods {
+			restarts = containersRestarts(pod.Status.ContainerStatuses)
+			if *restarts > 0 {
+				break
+			}
+		}
+		switch {
+		case k8sJob.Status.Failed > 0:
+			status = provision.StatusError
+		case k8sJob.Status.Succeeded > 0:
+			status = provision.StatusSucceeded
+		default:
+			status = provision.StatusStarted
 		}
 
-		createdAt := pod.CreationTimestamp.Time.In(time.UTC)
+		createdAt := k8sJob.CreationTimestamp.Time.In(time.UTC)
 		units = append(units, provision.Unit{
-			ID:        pod.Name,
-			Name:      pod.Name,
-			IP:        pod.Status.HostIP,
+			ID:        k8sJob.Name,
+			Name:      k8sJob.Name,
 			Status:    status,
```

Comment:
@morpheu what do you think about show IP of running pod?

Reply:
I omitted intentionally think on future scenarios where a job could spawn multiple pods

Reply:
Yeah! in case of multiple pods unique IP lose the sense